code/LeftAndMain.php

```diff
@@ -1441,18 +1441,8 @@ public function getEditForm($id = null, $fields = null)
         }
         $form->addExtraClass('fill-height');
 
-        // Add a default or custom validator.
-        if ($record->hasMethod('getCMSValidator')) {
-            $validator = $record->getCMSValidator();
-            // The clientside (mainly LeftAndMain*.js) rely on ajax responses
-            // which can be evaluated as javascript, hence we need
-            // to override any global changes to the validation handler.
-            if ($validator) {
-                $form->setValidator($validator);
-            }
-        } else {
-            $form->unsetValidator();
-        }
+        // A CompositeValidator is always available form a DataObject, but it may be empty (which is fine)
+        $form->setValidator($record->getCMSCompositeValidator());
```

Comment:
Whoops, I merged this too early =( we either need to check whether this method exists first, or bump the Composer constraint on framework so it always will. Would you mind doing one of those two things?

Reply:
Ah cool. Yea, I can do this now. I'll raise a new PR.

Reply:
PR raised here:
https://github.com/silverstripe/silverstripe-admin/pull/1080

I opted for adding BC support, rather than updating the constraint - I felt that was more forgiving if someone has pinned themselves to a particular framework version.